opencog/haskell/OpenCog/AtomSpace/Types.hs

```diff
@@ -0,0 +1,54 @@
+
+module OpenCog.AtomSpace.Types(
+    NodeType (..)
+  , LinkType (..)
+  , TruthValue (..)
+  , Atom (..)
+  , Link (..)
+  , Node (..)
+  , Handle (..)
```

Comment:
Atoms and Handles should be exactly the same thing. There should be no difference between atoms and handles.

There is a difference in the C++ code, but this is due to a terrible historical accident, and it will someday be fixed.

I guess you could say that a handle is like a pointer to an atom, but it would be best if you had an API that does not use pointers. Neither the scheme nor the python API's have pointers in them: you just work directly with the atoms that you want to work with.